projects/packages/my-jetpack/_inc/data/products/use-activate-plugins.ts

```diff
@@ -0,0 +1,90 @@
+import { useGlobalNotices } from '@automattic/jetpack-components';
+import { __, sprintf } from '@wordpress/i18n';
+import useAnalytics from '../../hooks/use-analytics';
+import { REST_API_SITE_PRODUCTS_ENDPOINT, QUERY_ACTIVATE_PRODUCT_KEY } from '../constants';
+import useSimpleMutation from '../use-simple-mutation';
+import { getMyJetpackWindowInitialState } from '../utils/get-my-jetpack-window-state';
+import useProducts from './use-products';
+import type { ProductCamelCase, ProductSnakeCase } from '../types';
+
+const setPluginActiveState = ( productId: string ) => {
+	const { items } = getMyJetpackWindowInitialState( 'products' );
+	if ( items[ productId ]?.plugin_slug === 'jetpack' ) {
+		return;
+	}
+	if ( items[ productId ]?.standalone_plugin_info.has_standalone_plugin ) {
+		window.myJetpackInitialState.products.items[
+			productId
+		].standalone_plugin_info.is_standalone_active = true;
+		window.myJetpackInitialState.products.items[
+			productId
+		].standalone_plugin_info.is_standalone_installed = true;
+	}
+};
+
+const getIsPluginAlreadyActive = ( detail: ProductCamelCase ) => {
+	const { standalonePluginInfo, isPluginActive } = detail;
+
+	if ( standalonePluginInfo?.hasStandalonePlugin ) {
+		return standalonePluginInfo?.isStandaloneActive;
+	}
+
+	return isPluginActive;
+};
+
+const useActivatePlugins = ( productIds: string[] ) => {
```

Comment:
I think a QOL improvement for developers would be to allow either the `string[]` or `string` type, and just convert the string to an array automatically if it is not already one. That way we don't have to pass in `[ slug ]` everywhere we activate only one plugin at a time. Same with the `useInstallPlugins` and `useProducts` hooks. Do you think that would be a good idea?

Reply:
Yes, thats a good idea, thanks! 👍